### PR TITLE
=doc #18636 Correction of mailbox documentation

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-docs/rst/scala/mailboxes.rst [hunk 1]
```diff
@@ -153,54 +153,57 @@ Akka comes shipped with a number of mailbox implementations:
   - Bounded: No
 
   - Configuration name: "akka.dispatch.UnboundedControlAwareMailbox"
+  
+* **UnboundedPriorityMailbox**
 
-Previously available mailbox implementations which were deprecated because they might block:
```
Comment (patriknw) [inline akka-docs/rst/scala/mailboxes.rst]: this was wrong

Comment (ktoso) [inline akka-docs/rst/scala/mailboxes.rst]: Thanks for looking into this @patriknw

File: akka-docs/rst/scala/mailboxes.rst [hunk 2]
```diff
@@ -208,7 +211,7 @@ Previously available mailbox implementations which were deprecated because they
 
   - FIFO order is preserved for messages of equal priority - contrast with the BoundedPriorityMailbox
 
-  - Blocking: Yes
+  - Blocking: Yes if used with non-zero ``mailbox-push-timeout-time``, otherwise No
```
Comment (ktoso) [inline akka-docs/rst/scala/mailboxes.rst]: right, good one

**Comment by ktoso** [comment]
LGTM

File: akka-docs/rst/scala/mailboxes.rst [hunk 3]
```diff
@@ -153,54 +153,57 @@ Akka comes shipped with a number of mailbox implementations:
...
 
-  - Configuration name: "akka.dispatch.UnboundedPriorityMailbox"
+  - Configuration name: "akka.dispatch.UnboundedStablePriorityMailbox"
 
-* **BoundedPriorityMailbox**
+Other bounded mailbox implementations which will block the sender if the capacity is reached and
+configured with non-zero ``mailbox-push-timeout-time``. 
 
-  - Backed by a ``java.util.PriorityQueue`` wrapped in an ``akka.util.BoundedBlockingQueue``
+.. note:: The following mailboxes should only be used with zero ``mailbox-push-timeout-time``.
```
Comment (patriknw) [inline akka-docs/rst/scala/mailboxes.rst]: It's probably difficult to see what I changed. I have moved the UnboundedPriorityMailbox  UnboundedStablePriorityMailbox up to the first list of mailboxes, and then comes this description about potentially blocking mailboxes and the list of such bounded mailboxes.

**Comment by ktoso** [comment]
LGTM, looked at the diff intensely :wink:

**Comment by ktoso** [comment]
Resolves #18636

**Pull request merged**